src/offset.js

```diff
@@ -82,27 +82,27 @@ jQuery.fn.extend({
 				});
 		}
 
-		var docElem, win,
+		var docElem, win, rect,
 			elem = this[ 0 ],
-			box = { top: 0, left: 0 },
 			doc = elem && elem.ownerDocument;
 
 		if ( !doc ) {
 			return;
 		}
 
-		docElem = doc.documentElement;
+		rect = elem.getBoundingClientRect();
 
-		// Make sure it's not a disconnected DOM node
-		if ( !jQuery.contains( docElem, elem ) ) {
-			return box;
+		// Make sure element is not hidden or disconnected
+		if ( !rect.width && !rect.height ) {
```

Comment:
> should this be || instead? as soon as a element has neither width nor height it is not visible, isn't it?

Honestly, I think it should be:
```javascript
if ( !rect.width && !rect.height && !rect.top && !rect.left && !rect.bottom && !rect.right ) {
```
Because only this case matches truly hidden elements (```display: none;``` or disconnected). In all other cases real values might be useful.